packages/backend/src/helpers/loadTimestamps.mjs

```diff
@@ -0,0 +1,17 @@
+export default async (objs, db) => {
+  const timestamps = await db.findMany('BlockTimestamp', {
+    where: {
+      number: objs.map(({ blockNumber }) => blockNumber),
+    },
+  })
+  const timestampsByBlock = timestamps.reduce((acc, o) => {
+    return {
+      [o.number]: o.timestamp,
+      ...acc,
+    }
+  }, {})
+  return objs.map((o) => ({
+    ...o,
+    timestamp: timestampsByBlock[o.blockNumber.toString()],
+  }))
+}
```

Comment:
if I understand correctly, these changes (all in PR) shouldn't be breaking anything used to render this..?

<img width="1254" alt="Screen Shot 2023-02-07 at 12 37 51 PM" src="https://user-images.githubusercontent.com/79813968/217347795-fc96a284-4fc1-418f-af4f-b690b166b9bb.png">

.. but now it's duplicating this user signup, and I've lost the timestamp.
It seems like something is going wrong in `ingestSignUps` in `unirep.js` but I haven't found it yet, maybe you can take a look?

also why do we want the timestamp as a string?

Reply:
The duplicate signups were coming from [here](https://github.com/Unirep/explorer/pull/19/files#diff-29abbe9ef6663638f95f2f1c4028581c4872c10ffcfaaa46df431ebd596c9f84R57). When ingesting new objects you need to make sure not to insert duplicates in the array.

The timestamp isn't taken as a string, the lint is `o.blockNumber.toString()`. Blocknumber is taken as a string because it's auto converted to a string when set to a key on the `byBlock` object.

Reply:
I also refactored the number handling. In the database we represent everything as decimal numbers. In the frontend we'll likely use hexadecimal so we'll need to handle converting.

Reply:
> The duplicate signups were coming from [here](https://github.com/Unirep/explorer/pull/19/files#diff-29abbe9ef6663638f95f2f1c4028581c4872c10ffcfaaa46df431ebd596c9f84R57). When ingesting new objects you need to make sure not to insert duplicates in the array.

I see it now thx.. I wanted to push a signup if the user had more than one in the incoming batch, I forgot about signups that would still be in there from a previous load.

do you know what is happening to the `timestamp`?  
<img width="907" alt="Screen Shot 2023-02-07 at 3 22 29 PM" src="https://user-images.githubusercontent.com/79813968/217380794-ffd5db8f-25b8-4236-ad64-e5c8d273c93e.png">
on the left is a signup object from the first log, the right is the same object from the second log- `timestamp` is missing
<img width="1058" alt="Screen Shot 2023-02-07 at 3 08 29 PM" src="https://user-images.githubusercontent.com/79813968/217380547-da3004ef-a947-46c6-9a61-ee02b8f18825.png">

Reply:
Yes, I changed how timestamps are fetched. I wrote a pretty primitive fetching system and there's a race condition where the timestamp sometimes isn't loaded before the event is processed. The timestamp fetching system also doesn't support retrying failed requests which is another problem.

I think i have a better approach to timestamps that i can implement here in a moment